boa_engine/src/builtins/date/mod.rs

```diff
@@ -1649,6 +1817,25 @@ impl Date {
         }
     }
 
+    /// `Date.prototype.toLocaleDateString()`
+    ///
+    /// The `toLocaleDateString()` method returns the date portion of the given Date instance
+    /// according to language-specific conventions.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-date.prototype.tolocaledatestring
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString
+    pub fn to_locale_date_string(
+        _this: &JsValue,
+        _args: &[JsValue],
+        _context: &mut Context,
+    ) -> JsResult<JsValue> {
+        unimplemented!()
```

Comment:
With this, right now if you execute `(new Date).toLocaleDateString()` boa panics. We should not add unimplemented methods that just panic.

Reply:
Ohh this was unfinished part and was in there because I asked for suggestion. As suggested, `locale*` are outside the scope, so I will remove it.

Reply:
> As suggested, `locale*` are outside the scope, so I should remove them, right?

Yes, let's either remove the functions or not add them to the object in `impl BuiltIn` for now. I think not having the function present on the object is the best way to not have the output interpreted false. Also it may be confusing in the conformance page when a few tests for the function pass but there is actually nothing implemented.